articles/techniques/testing-and-debugging.md

```diff
@@ -304,19 +304,19 @@ Wavefunction:
 1:      -0.5            -0.5
 ```
 
-In general, the state of a register that is entangled with another register is a mixed state rather than a pure state. In this case, @"microsoft.quantum.extensions.diagnostics.dumpregister" outputs the following message:
+In general, the state of a register that is entangled with another register is a mixed state rather than a pure state. In this case, <xref:microsoft.quantum.extensions.diagnostics.dumpregister> outputs the following message:
 
 ```
 Qubits provided (0;) are entangled with some other qubit.
 ```
 
-The following example shows you how you can use both @"microsoft.quantum.extensions.diagnostics.dumpregister" and @"microsoft.quantum.extensions.diagnostics.dumpmachine" in your Q# code:
+The following example shows you how you can use both <xref:microsoft.quantum.extensions.diagnostics.dumpregister> and <xref:microsoft.quantum.extensions.diagnostics.dumpmachine> in your Q# code:
 
 ```qsharp
 namespace app
 {
     open Microsoft.Quantum.Intrinsic;
-    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.extensions.Diagnostics;
```

Comment:
It looks like there's been a regression as of c62c7c18438b969273d7d95bf52171f75abdf117 here as well.
```suggestion
    open Microsoft.Quantum.Diagnostics;
```